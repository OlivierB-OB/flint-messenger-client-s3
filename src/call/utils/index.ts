export * from './defaultCallState';
export * from './getLocalInputs';
