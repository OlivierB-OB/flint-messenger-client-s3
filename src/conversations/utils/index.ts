export * from './consolidateUnseenMessages';
export * from './conversationComparator';
export * from './conversationFactory';
export * from './countUnseenMessages';
export * from './defaultConversationsState';
export * from './forgeNewConversationId';
export * from './lastMessageDate';
export * from './messageComparator';
