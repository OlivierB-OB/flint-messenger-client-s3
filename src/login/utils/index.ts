export * from './defaultLoginState';
export * from './validateRequiredField';
