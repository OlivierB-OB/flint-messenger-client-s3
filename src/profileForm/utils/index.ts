export * from './defaultProfileFormState';
export * from './validateConfirmationField';
export * from './validateEmailField';
export * from './validateNameField';
export * from './validatePasswordField';
