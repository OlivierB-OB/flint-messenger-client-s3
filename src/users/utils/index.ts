export * from './defaultUsersState';
export * from './userComparator';
