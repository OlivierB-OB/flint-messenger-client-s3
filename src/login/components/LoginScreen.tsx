import Container from '@material-ui/core/Container';
import Tab from '@material-ui/core/Tab';
import Tabs from '@material-ui/core/Tabs';
import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import { Action } from 'redux';
import { RegistrationForm } from '../../profileForm/components/RegistrationForm';
import { IProfileFormStatus } from '../../profileForm/types';
import { Loading } from '../../layout/components/Loading';
import { IAppState } from '../../appReducer';
import { ILoginStatus } from '../types';
import { LoginTabPanel } from './LoginTabPanel';
import { LoginForm } from './LoginForm';
import { resetProfileFormContent } from '../../profileForm/actions/resetProfileFormContent';
import { makeExitApplication } from '../../layout/actions/makeExitApplication';
import { ThunkDispatch } from 'redux-thunk';

export interface ILoginScreenDisplayProps {
  loginStatus: ILoginStatus;
  registrationStatus: IProfileFormStatus;
  resetRegistrationForm(): void;
}

export function LoginScreenDisplay({
  loginStatus,
  registrationStatus,
  resetRegistrationForm,
}: ILoginScreenDisplayProps) {
  const [tab, setTab] = useState(0);
  useEffect(() => void resetRegistrationForm(), [resetRegistrationForm]);
  if ([loginStatus, registrationStatus].includes('unavailable')) return <Loading />;
  return (
    <Container maxWidth="lg" style={{ marginTop: '2rem' }}>
      <Tabs
        value={tab}
        onChange={(_, newTab) => setTab(newTab)}
        indicatorColor="primary"
        textColor="primary"
        variant="fullWidth"
      >
        <Tab label="Login" />
        <Tab label="Register" />
      </Tabs>
      <LoginTabPanel value={tab} index={0}>
        <LoginForm />
      </LoginTabPanel>
      <LoginTabPanel value={tab} index={1}>
        <RegistrationForm />
      </LoginTabPanel>
    </Container>
  );
}

const mapStateToProps = ({ login, profileForm }: IAppState) => ({
  loginStatus: login.status,
  registrationStatus: profileForm.status,
});

const mapDispatchToProps = (dispatch: ThunkDispatch<IAppState, void, Action>) => ({
  exitApplication: () => dispatch(makeExitApplication()),
  resetRegistrationForm: () => dispatch(resetProfileFormContent()),
});

export const LoginScreen = connect(mapStateToProps, mapDispatchToProps)(LoginScreenDisplay);
