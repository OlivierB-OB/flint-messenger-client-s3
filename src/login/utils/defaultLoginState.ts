import { ILoginState } from '../types';
import { defaultStrField } from '../../profileForm/utils';

export function defaultLoginState(): ILoginState {
  return {
    status: 'ready',
    form: {
      email: defaultStrField(),
      password: defaultStrField(),
    },
  };
}
