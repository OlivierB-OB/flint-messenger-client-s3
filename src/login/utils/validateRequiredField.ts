import { IFormField } from '../../profileForm/types';

export function validateRequiredField(field: IFormField<string>): void {
  field.isValid = !!field.value;
  if (!field.isValid) field.error = 'this field is required';
  else delete field.error;
}
