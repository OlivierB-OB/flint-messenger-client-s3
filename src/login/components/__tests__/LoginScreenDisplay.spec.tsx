import Tabs from '@material-ui/core/Tabs';
import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow, mount } from 'enzyme';
import { LoginScreenDisplay } from '../LoginScreen';
import { defaultLoginState } from '../../utils';
import { defaultProfileFormState } from '../../../profileForm/utils';
import { Provider } from 'react-redux';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('LoginScreenDisplay', () => {
  it('should display the login form', async () => {
    const props: any = {
      loginStatus: 'login status',
      registrationStatus: 'registration status',
      resetRegistrationForm: jest.fn(),
    };
    const component = shallow(<LoginScreenDisplay {...props} />);
    expect(component).toMatchSnapshot();

    component.find(Tabs).simulate('change', undefined, 1);
    expect(component).toMatchSnapshot();
  });

  it('should display as loading if login status is unavailable', async () => {
    const props: any = {
      loginStatus: 'unavailable',
      registrationStatus: 'registration status',
      resetRegistrationForm: jest.fn(),
    };
    const component = shallow(<LoginScreenDisplay {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should display as loading if registration status is unavailable', async () => {
    const props: any = {
      loginStatus: 'login status',
      registrationStatus: 'unavailable',
      resetRegistrationForm: jest.fn(),
    };
    const component = shallow(<LoginScreenDisplay {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should reset the registration form on load', async () => {
    const props: any = {
      loginStatus: 'login status',
      registrationStatus: 'login status',
      resetRegistrationForm: jest.fn(),
    };

    const store = mockStore({
      login: defaultLoginState(),
      profileForm: defaultProfileFormState(),
    });

    mount(
      <Provider store={store}>
        <LoginScreenDisplay {...props} />
      </Provider>,
    );
    expect(props.resetRegistrationForm).toHaveBeenCalled();
  });
});
