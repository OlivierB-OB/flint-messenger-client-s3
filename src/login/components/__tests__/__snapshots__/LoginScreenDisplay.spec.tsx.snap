// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreenDisplay should display as loading if login status is unavailable 1`] = `<Loading />`;

exports[`LoginScreenDisplay should display as loading if registration status is unavailable 1`] = `<Loading />`;

exports[`LoginScreenDisplay should display the login form 1`] = `
<WithStyles(ForwardRef(Container))
  maxWidth="lg"
  style={
    Object {
      "marginTop": "2rem",
    }
  }
>
  <WithStyles(ForwardRef(Tabs))
    indicatorColor="primary"
    onChange={[Function]}
    textColor="primary"
    value={0}
    variant="fullWidth"
  >
    <WithStyles(ForwardRef(Tab))
      label="Login"
    />
    <WithStyles(ForwardRef(Tab))
      label="Register"
    />
  </WithStyles(ForwardRef(Tabs))>
  <LoginTabPanel
    index={0}
    value={0}
  >
    <Connect(LoginFormDisplay) />
  </LoginTabPanel>
  <LoginTabPanel
    index={1}
    value={0}
  >
    <Connect(RegistrationFormDisplay) />
  </LoginTabPanel>
</WithStyles(ForwardRef(Container))>
`;

exports[`LoginScreenDisplay should display the login form 2`] = `
<WithStyles(ForwardRef(Container))
  maxWidth="lg"
  style={
    Object {
      "marginTop": "2rem",
    }
  }
>
  <WithStyles(ForwardRef(Tabs))
    indicatorColor="primary"
    onChange={[Function]}
    textColor="primary"
    value={1}
    variant="fullWidth"
  >
    <WithStyles(ForwardRef(Tab))
      label="Login"
    />
    <WithStyles(ForwardRef(Tab))
      label="Register"
    />
  </WithStyles(ForwardRef(Tabs))>
  <LoginTabPanel
    index={0}
    value={1}
  >
    <Connect(LoginFormDisplay) />
  </LoginTabPanel>
  <LoginTabPanel
    index={1}
    value={1}
  >
    <Connect(RegistrationFormDisplay) />
  </LoginTabPanel>
</WithStyles(ForwardRef(Container))>
`;
